jupyter_server_fileid/manager.py

```diff
@@ -65,6 +67,8 @@ class BaseFileIdManager(ABC, LoggingConfigurable, metaclass=FileIdManagerMeta):
         config=True,
     )
 
+    con: Optional[sqlite3.Connection]
```

Comment:
I wonder if another base class needs to be injected here that contains SQLite-specifics, along with the "xxx_recursive()" methods.  Abstract base classes should be as minimal as possible where _any_ implementation must apply to _all_ derived classes and its clear that `con` would be an issue.

Reply:
Hm, yeah I agree that the base class is getting too bloated now. Putting SQLite-specific attributes and methods into a separate class (e.g. `SqliteMixin`) would allow File ID implementations to not use SQLite or implement migrations.

Reply:
This PR is growing too large as it is, let's just refactor this in a separate follow-up PR.